### PR TITLE
Update `overrides` section behavior in imports and inline

### PR DESCRIPTION
**Pull request by aknysh**
## what

- Update `overrides` section behavior in imports and inline
- Add acceptance tests for the `overrides` section for many scenarios (`overrides` at the stack level, team level, inline and in imports)
- Update docs
- Update `Go` to `1.24.0` and fix some issues in the code (`1.24.0` is more strict about the `Go` format functions always requiring the `format` as the first argument)
- Add unit tests

## why

- [Importing the overrides](https://atmos.tools/core-concepts/stacks/overrides/#importing-the-overrides) did not cover all possible scenarios (e.g. when importing the `overrides` for the imported components, the inline `overrides` for the same stack were not taken into account). This PR fixes this and adds acceptance tests for all scenarios
- Update the [Share Data Between Components](https://atmos.tools/core-concepts/share-data/) doc to describe how the `!terraform.output` YAML function can (and should be) used to to read the outputs ([remote state](https://atmos.tools/core-concepts/share-data/remote-state)) of components directly in Atmos stack manifests

## Summary by CodeRabbit

- **New Features**
  - Enhanced configuration processing distinguishes inline and imported overrides for greater deployment flexibility.
  - Added sample configuration files for various environments (development, production, staging, sandbox, and test) with team-specific settings.
  - Introduced new YAML function `!terraform.output` for improved data sharing in stack manifests.
  - New workflows added for testing various Atmos commands across multiple environments.

- **Documentation**
  - Updated documentation to showcase improved data sharing and new output functionalities.

- **Chores**
  - Upgraded Atmos CLI and related dependencies for enhanced performance.
  - Improved command output formatting for clearer error messages.
  - Added `.gitignore` entry to exclude specific directories from version control.

- **Tests**
  - Introduced test cases to validate configuration override behavior across environments.
  - Added tests for the `aboutCmd` and `supportCmd` commands to ensure correct output.
  - New test functions added for `ProcessYAMLConfigFile`, `ValidateStacks`, and `workflow` command to enhance coverage.
  - Comprehensive tests for markdown utility functions to ensure correct behavior.
  - New tests for handling invalid configurations and edge cases in YAML processing.
  - Added tests for vendor configuration handling and workflow execution.

### CODE REVIEW

File: cmd/support.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ var supportCmd = &cobra.Command{
 	SilenceUsage:       true,
 	SilenceErrors:      true,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		utils.PrintfMarkdown(supportMarkdown)
+		utils.PrintfMarkdown("%s", supportMarkdown)
```
Comment (osterman) [inline cmd/support.go:23]: Why are we changing this?

File: cmd/about.go [hunk 2]
```diff
@@ -17,7 +17,7 @@ var aboutCmd = &cobra.Command{
 	Long:  `Display information about Atmos, its features, and benefits.`,
 	Args:  cobra.NoArgs,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		utils.PrintfMarkdown(aboutMarkdown)
+		utils.PrintfMarkdown("%s", aboutMarkdown)
```
Comment (osterman) [inline cmd/about.go:20]: Why are we changing this?

File: internal/exec/stack_processor_utils.go [hunk 3]
```diff
@@ -196,13 +202,13 @@ func ProcessYAMLConfigFile(
 	// This is useful when generating Atmos manifests using other tools, but the imported files are not present yet at the generation time.
 	if err != nil {
 		if ignoreMissingFiles || skipIfMissing {
-			return map[string]any{}, map[string]map[string]any{}, map[string]any{}, map[string]any{}, map[string]any{}, nil
+			return map[string]any{}, map[string]map[string]any{}, map[string]any{}, map[string]any{}, map[string]any{}, map[string]any{}, map[string]any{}, nil
 		} else {
-			return nil, nil, nil, nil, nil, err
+			return nil, nil, nil, nil, nil, nil, nil, err
```
Comment (osterman) [inline internal/exec/stack_processor_utils.go:207]: This is getting out of hand with 8 return values. We should move to an object.

File: pkg/utils/markdown_utils.go [hunk 4]
```diff
@@ -78,10 +80,12 @@ func PrintErrorMarkdownAndExit(title string, err error, suggestion string) {
 	os.Exit(1)
```
Comment (osterman) [inline pkg/utils/markdown_utils.go:80]:
```suggestion
        // TODO: Refactor so that we only call `os.Exit` in `main()` or `init()` functions. 
        // Exiting here makes it difficult to test.
        // revive:disable-next-line:deep-exit
	os.Exit(1)
```

**Review by osterman — APPROVED** [review]
